test/unit/conditions/base/timelock.test.ts

```diff
@@ -0,0 +1,39 @@
+import { TimelockCondition } from '../../../../src/conditions/base';
+
+describe('validation', () => {
+  const returnValueTest = {
+    index: 0,
+    comparator: '>',
+    value: '100',
+  };
+
+  it('accepts a valid schema', () => {
+    const timelock = new TimelockCondition({
+      returnValueTest,
+    });
+    expect(timelock.toObj()).toEqual({
+      returnValueTest,
+      method: 'timelock',
+    });
+  });
+
+  it('rejects an invalid schema', () => {
+    const badTimelockObj = {
+      // Intentionally replacing `returnValueTest` with an invalid test
+      returnValueTest: {
+        ...returnValueTest,
+        comparator: 'not-a-comparator',
+      },
+    };
+
+    const badTimelock = new TimelockCondition(badTimelockObj);
+    expect(() => badTimelock.toObj()).toThrow(
```

Comment:
Given this test - any idea how this slipped through, https://discord.com/channels/866378471868727316/870383717418410066/1107101659399729192? Just curious really.

Reply:
Nothing stops anyone from breaking this code really. You can always go `{randomJson: true} as Condition` or hardcode some values into Porter client. It's best we ask them for a code example to reproduce this.